library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -998,6 +1044,34 @@ public void setMaxRetryCount(int maxRetryCount)
   }
 
   /**
+   * This class scans a directory and put the list of files in a blocking queue.
+   */
+  public class DirectoryScannerThread implements Runnable
+  {
+    @Override
+    public void run()
+    {
+      try {
+        while (execute) {
+          if (pendingFiles.isEmpty() && fileQueue.isEmpty() && (System.currentTimeMillis() - scanIntervalMillis >= lastScanMillis)) {
+            LinkedHashSet<Path> newFileSet = scanner.scan(fs, filePath, processedFiles);
+            for (Path newFile : newFileSet) {
+              fileQueue.put(newFile);
+            }
+
+            lastScanMillis = System.currentTimeMillis();
+          } else {
+            // Thread.sleep(DEFAULT_SLEEP_TIME);
+            Thread.sleep(scanIntervalMillis);
+          }
+        }
+      } catch (Exception e) {
+        execute = false;
```

Comment:
In case of Exception, we should notify the main thread and take appropriate action there. Take a look at 
https://github.com/apache/apex-malhar/blob/master/library/src/main/java/com/datatorrent/lib/io/fs/FileSplitterInput.java#L163 and
https://github.com/apache/apex-malhar/blob/master/library/src/main/java/com/datatorrent/lib/io/fs/FileSplitterInput.java#L383